homeassistant/components/switcher_kis/button.py

```diff
@@ -0,0 +1,159 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitcherDataUpdateCoordinator
+from .const import SIGNAL_DEVICE_ADD
+from .utils import get_breeze_remote_manager
+
+
+@dataclass
+class SwitcherThermostatButtonDescriptionMixin:
+    """Mixin to describe a Switcher Thermostat Button entity."""
+
+    press_fn: Callable[[SwitcherType2Api, SwitcherBreezeRemote], SwitcherBaseResponse]
+    supported: Callable[[SwitcherBreezeRemote], bool]
+
+
+@dataclass
+class SwitcherThermostatButtonEntityDescription(
+    ButtonEntityDescription, SwitcherThermostatButtonDescriptionMixin
+):
+    """Class to describe a Switcher Thermostat Button entity."""
+
+
+THERMOSTAT_BUTTONS = [
+    SwitcherThermostatButtonEntityDescription(
+        key="assume_on",
+        name="Assume on",
+        icon="mdi:fan",
+        entity_category=EntityCategory.CONFIG,
+        press_fn=lambda api, remote: api.control_breeze_device(
+            remote, state=DeviceState.ON, update_state=True
+        ),
+        supported=lambda remote: bool(remote.on_off_type),
+    ),
+    SwitcherThermostatButtonEntityDescription(
+        key="assume_off",
+        name="Assume off",
+        icon="mdi:fan-off",
+        entity_category=EntityCategory.CONFIG,
+        press_fn=lambda api, remote: api.control_breeze_device(
+            remote, state=DeviceState.OFF, update_state=True
+        ),
+        supported=lambda remote: bool(remote.on_off_type),
+    ),
+    SwitcherThermostatButtonEntityDescription(
+        key="vertical_swing_on",
+        name="Vertical swing on",
+        icon="mdi:autorenew",
+        press_fn=lambda api, remote: api.control_breeze_device(
+            remote, swing=ThermostatSwing.ON
+        ),
+        supported=lambda remote: bool(remote.separated_swing_command),
+    ),
+    SwitcherThermostatButtonEntityDescription(
+        key="vertical_swing_off",
+        name="Vertical swing off",
+        icon="mdi:autorenew-off",
+        press_fn=lambda api, remote: api.control_breeze_device(
+            remote, swing=ThermostatSwing.OFF
+        ),
+        supported=lambda remote: bool(remote.separated_swing_command),
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Switcher button from config entry."""
+
+    @callback
+    async def async_add_buttons(coordinator: SwitcherDataUpdateCoordinator) -> None:
```

Comment:
A coroutine function is never a callback.

Reply:
That is a leftover since I started with a `sync` code, I will remove it